### PR TITLE
fix(loader): support file name with prefix for hdfs source

### LINKED ISSUE #570: [Improve] Loader support file name with prefix for hdfs source
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-hugegraph-toolchain/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Feature Description (功能描述)

Loader support file name with prefix for hdfs source

when read data warehouse file, There will be some _success empty files, etc., and you need to specify the file name prefix.

struct.json

```JavaScript
{
    "vertices":[
        {
            "label":"user",
            "input":{
                "type":"hdfs",
                "path":"hdfs:///vertex/louvain_user_vertex/part-*",
                "compression":"PARQUET",
                "core_site_path":"/conf/core-site.xml",
                "hdfs_site_path":"/conf/hdfs-site.xml",
                "header":[
                    "user_id"
                ],
                "charset":"UTF-8",
                "skipped_line":{
                    "regex":"(^#|^//).*"
                }
            },
            "id":"user_id",
            "null_values":[
                "NULL",
                "null",
                ""
            ]
        }
    ]
}

```

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by JackyYangPassion**
## Purpose of the PR

- close #570  

## Main Changes

support file name with prefix for hdfs source

## Verifying these changes

- [ ] Trivial rework / code cleanup without any test coverage. (No Need)
- [ ] Already covered by existing tests, such as *(please modify tests here)*.
- [x] Need tests and can be verified as follows:
    - xxx

## Does this PR potentially affect the following parts?

- [x]  Nope
- [ ]  Dependencies (add/update license info) 
- [ ]  Modify configurations
- [ ]  The public API
- [ ]  Other affects (typed here)

## Documentation Status

- [x]  `Doc - TODO` 
- [ ]  `Doc - Done` 
- [ ]  `Doc - No Need`

### CODE REVIEW

File: hugegraph-loader/src/main/java/org/apache/hugegraph/loader/reader/hdfs/HDFSFileReader.java [hunk 1]
```diff
@@ -109,7 +125,18 @@ protected List<Readable> scanReadables() throws IOException {
...
+                statuses = this.hdfs.listStatus(path);
+            } else {
+                PathFilter prefixFilter = new PathFilter() {
+                    @Override
+                    public boolean accept(Path path) {
+                        return path.getName().startsWith(prefix);
+                    }
+                };
+                statuses = this.hdfs.listStatus(path,prefixFilter);
+            }
```
Comment (imbajin) [inline hugegraph-loader/src/main/java/org/apache/hugegraph/loader/reader/hdfs/HDFSFileReader.java:139]: maybe we could not list all HDFS_Path together, use iterator to scan it?

Add a commit here
```java
            assert this.hdfs.isDirectory(path);
            FileStatus[] statuses;
            if (prefix == null || prefix.isEmpty()) {
                statuses = this.hdfs.listStatus(path);
            } else {
                PathFilter prefixFilter = path1 -> path1.getName().startsWith(prefix);
                statuses = this.hdfs.listStatus(path, prefixFilter);
            }
            Path[] subPaths = FileUtil.stat2Paths(statuses);
            for (Path subPath : subPaths) {
                if (filter.reserved(subPath.getName())) {
                    paths.add(new HDFSFile(this.hdfs, subPath));
                }
            }

```

Comment (JackyYangPassion) [reply hugegraph-loader/src/main/java/org/apache/hugegraph/loader/reader/hdfs/HDFSFileReader.java]: The HDFSFileReader supports filename prefixes just for the convenience of data warehouse edge table loading, so I believe it's better to stick to a fixed usage and not recommend multiple ways of writing the path.

File: hugegraph-loader/src/main/java/org/apache/hugegraph/loader/reader/hdfs/HDFSFileReader.java [hunk 2]
```diff
@@ -62,7 +65,20 @@ public HDFSFileReader(HDFSSource source) {
         } catch (IOException e) {
             throw new LoadException("Failed to create HDFS file system", e);
         }
-        Path path = new Path(source.path());
+        String input = source.path();
+        if (input.contains("*")) {
+            int lastSlashIndex = input.lastIndexOf('/');
+            if (lastSlashIndex != -1) {
+                input_path = input.substring(0, lastSlashIndex);
+                prefix = input.substring(lastSlashIndex + 1, input.length() - 1);
```
[the remark this answers is not in the document]
Comment (JackyYangPassion) [reply hugegraph-loader/src/main/java/org/apache/hugegraph/loader/reader/hdfs/HDFSFileReader.java]: add  unit test for HDFS Load

[the remark this answers is not in the document]
Comment (JackyYangPassion) [reply hugegraph-loader/src/main/java/org/apache/hugegraph/loader/reader/hdfs/HDFSFileReader.java]: Data warehouse files have a characteristic of having a unified prefix, such as part*, so I believe there's no need to consider regular expression matching in the scenarios adapted to this PR.

**Review by imbajin — APPROVED** [review]
Get it, keep the logic simple now

**Pull request merged**